db/compaction/compaction_picker.cc

```diff
@@ -1040,6 +1042,13 @@ Status CompactionPicker::SanitizeCompactionInputFilesForAllLevels(
       }
     }
   }
+
+  if (!valid_input_files_found) {
+    return Status::Aborted(
+        "There are no valid input files found for the specified files.");
+  }
+  assert(!smallestkey.empty());
```

Comment:
I think empty user key is allowed.

Reply:
I didn't quite get this, in  the `RangeOverlapWithCompaction` function, empty user key can cause assertions in this logic: https://github.com/facebook/rocksdb/blob/9bddac0dcfb6ea7fc55e64f41a2670cbc23c4a2d/db/compaction/compaction_picker.cc#L266

So I think it's better to not pass empty boundaries to that function yet. These user key boundaries come from the existing files' boundaries. At this point, they should either represent some file's actual range, or no valid files found and they are empty. So here we want to not check further if range overlap when it's empty.

Reply:
Good catch! I didn't get that you meant `Put("","")` is allowed. Thanks for catching this!